### PR TITLE
fix: add url to sp event

### PR DESCRIPTION
**Pull request by icazevedo**
## What's the purpose of this pull request?

This PR solves blank intelligent search reports

## How it works?

Adds missing `url` field on SP event using the referrer as the value

## How to test it?

On it

### Starters Deploy Preview

On it

### CODE REVIEW

File: packages/api/src/platforms/vtex/resolvers/query.ts [hunk 1]
```diff
@@ -147,12 +148,15 @@ export const Query = {
...
 
     const after = maybeAfter ? Number(maybeAfter) : 0
-    const searchArgs = {
+    const referrer = ctx.headers.referrer
+
+    const searchArgs: Omit<SearchArgsWithEventArgs, 'type'> = {
       page: Math.ceil(after / first),
       count: first,
-      query,
+      query: query ?? undefined,
```
Comment (eduardoformiga) [inline packages/api/src/platforms/vtex/resolvers/query.ts:156]: is that for handling `null` values?

**Review by eduardoformiga — APPROVED** [review]

**Review by eduardoformiga — COMMENTED** [review]
Could you update the tests please?

**Comment by icazevedo** [comment]
> Could you update the tests please?

I don't think it affects any current tests, but I surely can write them. I'll deploy this and then write them due to the urgency of the matter.

File: packages/api/src/platforms/vtex/resolvers/query.ts [hunk 2]
```diff
@@ -147,12 +148,15 @@ export const Query = {
     }
 
     const after = maybeAfter ? Number(maybeAfter) : 0
-    const searchArgs = {
+    // const referer = ctx.headers.referer
```
Comment (emersonlaurentino) [inline packages/api/src/platforms/vtex/resolvers/query.ts:151]: 👀

Comment (icazevedo) [reply packages/api/src/platforms/vtex/resolvers/query.ts:151]: 🔧

**Pull request closed without merging**